packages/runtime/container-runtime/src/connectionTelemetry.ts

```diff
@@ -258,6 +258,56 @@ class OpPerfTelemetry {
         }
     }
 }
+export interface IPerfSignalReport {
+    /**
+     * Identifier for the signal being submitted in order to
+     * allow collection of data around the roundtrip of signal messages.
+     */
+    signalSequenceNumber: number;
+    /**
+     * Number of signals that were expected but not received.
+     */
+    signalsLost: number;
+}
+
+export class SignalTimestampCache {
```

Comment:
I think this class needs some tests as it does some very cool logic.

Reply:
I will add them - just waiting for Vlad to take a look to make sure this solution seems reasonable on his end.